spring-cloud-stream-binders/spring-cloud-stream-binder-test/src/test/java/org/springframework/cloud/stream/binder/MessageChannelBinderSupportTests.java

```diff
@@ -112,6 +112,19 @@ public void testString() throws IOException {
 	}
 
 	@Test
+	public void testStringXML() throws IOException {
+		Message<?> message = MessageBuilder
+				.withPayload("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?><test></test>")
+				.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.TEXT_XML)
+				.build();
+		Message<?> converted = binder.serializePayloadIfNecessary(message).toMessage();
+		assertEquals(MimeTypeUtils.TEXT_PLAIN, contentTypeResolver.resolve(converted.getHeaders()));
```

Comment:
This definitely goes beyond this simple PR, but this really seems weird to me.
IMHO, when "serialization" happens, there should either
* be no more concept of contentType, as everything "is just bytes"
* or if there is the concept of contentType, the original contentType should apply, especially when it has been explicitly set (as opposed to inferred).

I'd be curious to hear what @dsyer @markfisher and @mbogoevici think about this, but this should not block this PR I guess